core/src/commonMain/kotlin/entity/Member.kt

```diff
@@ -30,9 +30,9 @@ public class Member(
         get() = memberData.guildId
 
     /**
-     * The name as shown in the discord client, prioritizing the [nickname] over the [username].
+     * The name as shown in the discord client, prioritizing [nickname] over [globalName] and [username].
      */
-    public val displayName: String get() = nickname ?: username
+    override val displayName: String get() = nickname ?: super.displayName
```

Comment:
should this really be overridden? or should `User` have an extension property for this?

i.e. what would you expect in the following situation:
```kt
fun doSomethingWithUser(user: User) {
    println(user.displayName)
}

val member: Member = TODO()
// should this print the member nickname or the user globalName?
doSomethingWithUser(member)
```

Reply:
This is not alright as the user will lose their ability to pick the original display name.
make a effectiveDisplayName property on the member 
otherwise you can always pick the displayName property

Reply:
so you suggest changing the behavior of the already existing `Member.displayName` like this?
```kt
// new property in the User class:
val displayName = globalName ?: username

// previous behavior in the Member class:
val displayName = nickname ?: username
// new behavior in the Member class:
val displayName = globalName ?: username // same as for User

// new property in the Member class:
val effectiveDisplayName = nickname ?: globalName ?: username
```

Reply:
@lukellmann Yes, that's the point
Since the global name is prioritized over the username I think it would serve as a better approach in the new username system.

Reply:
i actually came up with a way that wouldn't change the behavior for members but instead deprecate it